src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs

```diff
@@ -81,12 +81,28 @@ public OpenAIResponsesChatClient(OpenAIResponseClient responseClient)
     public async Task<ChatResponse> GetResponseAsync(
         IEnumerable<ChatMessage> messages, ChatOptions? options = null, CancellationToken cancellationToken = default)
     {
-        _ = Throw.IfNull(messages);
+        var inputMessages = Throw.IfNull(messages) as IReadOnlyCollection<ChatMessage> ?? [.. messages];
 
-        // Convert the inputs into what OpenAIResponseClient expects.
-        var openAIResponseItems = ToOpenAIResponseItems(messages, options);
         var openAIOptions = ToOpenAIResponseCreationOptions(options);
 
+        // Convert the inputs into what OpenAIResponseClient expects.
+        var openAIResponseItems = ToOpenAIResponseItems(inputMessages, options);
```

Comment:
Why was it necessary to change the type of inputMessages? Couldn't L94 below fail iff openAIResponseItems.Count > 0? Or what scenario has zero input messages but > 0 converted items?